### PR TITLE
[INLONG-5133][Sort] Support InfluxDB extract node

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Prepare a Pull Request
- Fixes https://github.com/apache/inlong/issues/5133

### Motivation

Add Apache InfluxDB Extract for Sort

### Modifications

The details can be found at https://github.com/apache/inlong/issues/5133.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by liangyepianzhou** [comment]
@jun0315 I have finished writing the main framework of influx source, and now there are three points that need to be improved. I've listed it below, please take a look at this question when you have time.
1. Parameters
    * The parameters written in the building function in the `InfluxDBSource` may not be complete. So if you need to add parameters here, you also need to add them in the `getScanRuntimeProvider` and `InfluxDBTableSourceFactory`.
    * Divide required and optional parameters into the methods `requiredOptions` and `optionalOptions`
2. Serialization
   * InfluxDB Table Source needs to pass in a `deserializer DebeziumDeserializationSchema<RowData>` deserializer when calling the function;  This can get help at [flink-connector-influxdb2](https://github.com/apache/bahir-flink/tree/master/flink-connector-influxdb2)
3. Connecter CanonicalName: There is a configuration item in InfluxDBSource that I don't know how to deal with
    `  props.setProperty("connector.class", InfluxDbConnector.class.getCanonicalName());`

In order to better understand the workflow of the Flink influx DB source, I wrote a simple process described below:
Workflow of Flink source connecter (simplified version):
`InfluxDBTableSourceFactory` creates an `InfluxDBTableSource`, and the `getScanRuntimeProvider` method of `InfluxDBTableSource` creates a `DebeziumSourceFunction` responsible for fetching captured change data from the database into Flink.
And InfluxDBSource is similar to a creator used to create a DebeziumSourceFunction.
And more details can be found [here](https://nightlies.apache.org/flink/flink-docs-release-1.14/zh/docs/dev/table/sourcessinks/).

**Comment by liangyepianzhou** [comment]
@dockerzhang Please help check this PR when you have time, THX.
In addition, I noticed that except for MySQL and JDBC, other CDCs only implement source and not sink. Influx DB also only needs to implement source, right?

**Comment by EMsnap** [comment]
> @dockerzhang Please help check this PR when you have time, THX. In addition, I noticed that except for MySQL and JDBC, other CDCs only implement source and not sink. Influx DB also only needs to implement source, right?

I guess this issue means you need to implement both sink and source

**Comment by EMsnap** [comment]
> > @dockerzhang Please help check this PR when you have time, THX. In addition, I noticed that except for MySQL and JDBC, other CDCs only implement source and not sink. Influx DB also only needs to implement source, right?
> 
> I guess this issue means you need to implement both sink and source

also the sink of these connector is implement by jdbc driver, can check the jdbc connector for details

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/Node.java [hunk 1]
```diff
@@ -75,6 +76,7 @@
         @JsonSubTypes.Type(value = TubeMQExtractNode.class, name = "tubeMQExtract"),
         @JsonSubTypes.Type(value = RedisExtractNode.class, name = "redisExtract"),
         @JsonSubTypes.Type(value = DorisExtractNode.class, name = "dorisExtract"),
+        @JsonSubTypes.Type(value = InfluxDBExtractNode.class, name = "influxExtract"),
```
Comment (yunqingmoswu) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/Node.java:79]: Please keep the naming consistent of `influxExtract`  between the Node and ExtractNode.

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/ExtractNode.java [hunk 2]
```diff
@@ -62,7 +63,8 @@
         @JsonSubTypes.Type(value = TubeMQExtractNode.class, name = "tubeMQExtract"),
         @JsonSubTypes.Type(value = PulsarExtractNode.class, name = "pulsarExtract"),
         @JsonSubTypes.Type(value = RedisExtractNode.class, name = "redisExtract"),
-        @JsonSubTypes.Type(value = DorisExtractNode.class, name = "dorisExtract")
+        @JsonSubTypes.Type(value = DorisExtractNode.class, name = "dorisExtract"),
+        @JsonSubTypes.Type(value = InfluxDBExtractNode.class, name = "influxDBExtract")
```
Comment (yunqingmoswu) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/ExtractNode.java:67]: Please keep the naming consistent of influxExtract between the Node and ExtractNode.

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/InfluxDBExtractNode.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Extract node for influxDB, note that InfluxDB should work in replicaSet mode
+ */
+@EqualsAndHashCode(callSuper = true)
+@JsonTypeName("influxDBExtract")
```
Comment (yunqingmoswu) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/InfluxDBExtractNode.java:43]: Please keep the naming consistent of influxExtract between the Node and ExtractNode.

**Review by yunqingmoswu — COMMENTED** [review]
Please add some unit test for node protocol and connector.

**Pull request closed without merging**